File: server/spring/src/main/java/com/ssafy/tadak/spring/common/exception/dto/ErrorResponse.java
package com.ssafy.tadak.spring.common.exception.dto;

import com.ssafy.tadak.spring.common.exception.ErrorCode;

public record ErrorResponse (
        String code,
        String message
){
    public static ErrorResponse from(ErrorCode errorCode) {
        return new ErrorResponse(errorCode.code(), errorCode.message());
    }
}

File: server/auth_gateway/src/main/kotlin/exception/dto/ErrorResponse.kt
package com.tadak.exception.dto

import com.tadak.exception.ErrorCode
import kotlinx.serialization.Serializable

@Serializable
data class ErrorResponse(
    val code: String,
    val message: String
) {
    companion object {
        fun from(errorCode: ErrorCode): ErrorResponse {
            return ErrorResponse(
                code = errorCode.code,
                message = errorCode.message
            )
        }
    }
}